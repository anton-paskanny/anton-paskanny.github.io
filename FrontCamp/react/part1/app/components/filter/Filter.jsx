import React from 'react';

import './styles.css'

export default (props) => (
  <div className="filter">
    <h3 className="filter__title">
      Filter blogs by author's name
    </h3>
    <input type="text"
           placeholder="Author's name"
           onChange={props.filterByAuthor}
           className="filter__input"
           disabled={props.blogsLength === 0}
    />
    { props.showFilteringError && <p className="filter__error">Nothing was found</p> }
  </div>
)
