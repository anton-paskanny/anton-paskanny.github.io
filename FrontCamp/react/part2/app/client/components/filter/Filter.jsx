import React from 'react';

import './styles.css'


export default class Filter extends React.Component {
  constructor(props) {
    super(props);
    this.searchInput = {};
  }
  render() {
    return (
      <div className="filter">
        <h3 className="filter__title">
          Filter blogs by author's name
        </h3>
        <input type="text"
               placeholder="Author's name"
               onChange={this.props.filterByAuthor}
               className="filter__input"
               ref={input => this.searchInput = input}
        />
        {
          !this.props.blogsLength &&
          this.searchInput.value &&
          <p className="filter__error">Nothing was found</p>
        }
      </div>
    )
  }
}
