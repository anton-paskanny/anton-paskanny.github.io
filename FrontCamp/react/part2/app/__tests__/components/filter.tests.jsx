import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, configure, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';

import Filter from '../../client/components/filter/Filter';

configure({ adapter: new Adapter() });

describe('Filter component', () => {

  test('should render Filter component', () => {
    const component = shallow(<Filter />);
    expect(component).toHaveLength(1);
  });

  test('should render correctly', () => {
    const component = renderer.create(<Filter />).toJSON();
    expect(component).toMatchSnapshot();
  });

  test('should has necessary filterByAuth and blogsLength in props', () => {
    const props = {
      filterByAuthor: jest.fn(),
      blogsLength: 3
    };

    const component = shallow(<Filter {...props} />);

    expect(component.instance().props.filterByAuthor).toBeTruthy();
    expect(component.instance().props.blogsLength).toBeTruthy();
  });
});
