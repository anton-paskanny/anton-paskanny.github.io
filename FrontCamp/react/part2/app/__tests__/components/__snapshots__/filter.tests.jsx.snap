// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component should render correctly 1`] = `
<div
  className="filter"
>
  <h3
    className="filter__title"
  >
    Filter blogs by author's name
  </h3>
  <input
    className="filter__input"
    onChange={undefined}
    placeholder="Author's name"
    type="text"
  />
</div>
`;
